### PR TITLE
ci(*): add image build and publish

### PR DESCRIPTION
**Pull request by vdice**
* adds the Dockerfile for the `gh-release` image used by the `GitHubRelease` job
* re-orgs to put all images into the `images` directory
* adds CI
* bumps a few (dev) deps per yarn audit

### CODE REVIEW

File: brigade.js [hunk 1]
```diff
@@ -28,3 +45,7 @@ events.on("check_suite:rerequested", runSuite);
 
 events.on("issue_comment:created", (e, p) => Check.handleIssueComment(e, p, runSuite));
 events.on("issue_comment:edited", (e, p) => Check.handleIssueComment(e, p, runSuite));
+
+events.on("push", (e, p) => {
```
Comment (radu-matei) [inline brigade.js:93]: So on `check_suite:requested` we build the Node library, and on a `push` we publish the images?
Isn't the `push` event triggered even on branch pushes? If so, we end up executing both on all branch pushes as well.

Comment (vdice) [reply brigade.js:93]: That's correct.  Note that the `check_suite` events will be triggered on all Pull Requests.  Currently, I figured we wouldn't want to build and publish these images for every pull request.  However, we could do so.  Or, perhaps a good compromise would be to build the images for all checks (PRs) but only push/publish images in response to the `push` event?

**Comment by vdice** [comment]
@radu-matei could you revisit when you have a chance?

**Review by radu-matei — APPROVED** [review]

**Comment by vdice** [comment]
(Last commits = rebase with master and then audit fix.)

**Pull request merged**